internal/controller/nonadminbackup_controller.go

```diff
@@ -118,6 +126,52 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
+// DeleteVeleroBackup deletes original Velero Backup object when the deleteVeleroBackup Spec is set to true
+//
+// Parameters:
+//
+//	ctx: Context for the request.
+//	logrLogger: Logger instance for logging messages.
+//	nab: Pointer to the NonAdminBackup object.
+func (r *NonAdminBackupReconciler) DeleteVeleroBackup(ctx context.Context, logrLogger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (exitReconcile bool, errorReconcile error) {
+	rLog := log.FromContext(ctx)
+	logger := logrLogger.WithValues("DeleteVeleroBackup", nab.Namespace)
+
+	// Delete Velero Backup if the deleteVeleroBackup was set to true
+	if nab.Spec.DeleteVeleroBackup != nil && *nab.Spec.DeleteVeleroBackup {
+		deleted, deleteErr := function.DeleteVeleroBackup(ctx, r.Client, rLog, nab)
+		if !deleted {
+			// The Velero Object was not found
+			// Remove NonAdminBackup
+			// There is potential that Velero Sync controller will recreate Velero Backup
+			// if one exists in the s3 storage, and this will as well cause NonAdminBackup
+			// to be recreated by the NAC sync controller, but at this stage we have current
+			// proper state reflected.
+			logger.V(1).Info(fmt.Sprintf("Velero Backup not deleted for NAB: %s", nab.Name))
```

Comment:
would be interesting to add reason here?

something like `"Velero Backup %s, referenced in for %s, was not found in %s namespace and was not deleted"`

Reply:
so this will cause NAB to be in phase Backing Off, and all the info about the referenced velero backup is in the status of that NAB, do we want to duplicate this info within debug logs?